### PR TITLE
ARROW-17222: [Docs][Archery][Integration] Document the current Integration test cases covered by archery

### PR DESCRIPTION
**Pull request by zeroshade**
Suggested by @nealrichardson, it makes sense to have a documented section of what test cases are covered by Archery's integration tests so we know what is actually covered or not for future development. This will make it easier to make changes to the format in the future if necessary as we'll know what (if any) additional testing will need to be done and added to Archery.

### CODE REVIEW

File: docs/source/format/Integration.rst [hunk 1]
```diff
@@ -396,3 +396,104 @@ of ``listSize`` 4, then the data inside the "children" of that ``FieldData``
 will have count 28.
 
 For "null" type, ``BufferData`` does not contain any buffers.
+
+Current Archery Integration Test Cases
+--------------------------------------
+
+For ease of future development, it makes sense to document the current test
+cases which are covered by the :ref:`Archery <archery>` integration tests.
+This can serve to make it easy to understand what manual testing may need to
```
Comment (assignUser) [inline docs/source/format/Integration.rst:405]:
```suggestion
This list can make it easier to understand what manual testing may need to
```
I don't we need to justify why we document our tools 😁

**Review by assignUser — COMMENTED** [review]
Looks good, thanks! Just a few nitty comments.

One thing I noticed while looking this over is that the explanation for what happens during the tests is a bit ambiguous imh:
It is clear that each implementation converts the json to arrow binary and
> The test executable is also capable of validating the contents of a binary
  file against a corresponding JSON file

The "it is capable" is a bit vague. I am guessing that each implementation verifies the binary arrow file generated by the other implementations vs the JSON.  I think a small change could make this much clearer :)

**Comment by zeroshade** [comment]
@assignUser Thanks for the suggestions! What do you think of my last commit which attempts to clarify the ambiguity you brought up?

**Review by assignUser — APPROVED** [review]
Nice, thanks!

**Pull request merged**